src/App.js

```diff
@@ -1,10 +1,51 @@
+import React, { useEffect, useState, useCallback } from 'react';
+
+import GlobalStyle from './styles/GlobalStyle';
+import { Main, Section } from './styles/Container';
+import { getLocalStorage, syncLocalStorage } from './components/TodoState';
+import { runToast, ToastComponent } from './components/toast/Toast';
+import InputTodo from './components/InputTodo';
+import TodoItemList from './components/TodoItemList';
 
 function App() {
-  return (
-    <div>
-      <h1>15기 화이팅~</h1>
-    </div>
-  );
+    const [list, setList] = useState(getLocalStorage);
+
+    // list 상태 변화할 때마다 로컬스토리지 동기화
+    useEffect(() => syncLocalStorage(list), [list]);
+
+    const toggleTodo = useCallback(
+        (id) => {
+            setList((list) => list.map((item) => (item.id === id ? { ...item, done: !item.done } : item)));
```

Comment:
arrow function이 불필요하게 들어간 감이 있네요.
```suggestion
            setList(list.map((item) => (item.id === id ? { ...item, done: !item.done } : item)));
```

Reply:
리액트를 다루는 기술 책을 보고 참고를 했는데 이 부분에 대해선 조금더 공부가 필요할것 같습니다ㅠㅠ 감사합니다!!

Reply:
선종 님께서 말씀하신 대로 useCallback의 dependency array에 list를 추가하시면 함수형 업데이트를 사용하지 않으셔도 됩니다!

또는 지금 작성해 주신 것처럼 함수형 업데이트를 사용하신다면 list를 참조하지 않으니 dependency array에 추가할 필요도 없고 최적화하기 더 좋은 걸로 알고 있어요!